### ISSUE TITLE
DNS resolving doesn't work with systemd-resolved on the host

### ISSUE THREAD

**Issue by iaguis** [kinvolk/kube-spawn#128]
When testing on Ubuntu 17.04 which uses `systemd-resolved` , DNS didn't work on the nodes.

In that setup, `systemd-resolved` [sets up a local DNS stub listener](https://www.freedesktop.org/software/systemd/man/systemd-resolved.service.html#Description) that routes requests to `systemd-resolved`. This means `/etc/resolv.conf` contains this line

```
nameserver 127.0.0.53
```

The containers will also have that line but in them there will be nothing listening on that address because the network namespace is different from the one on the host, so DNS requests fail.

It seems more people have problems with this: https://github.com/systemd/systemd/issues/3649#issuecomment-317856930

**Comment by iaguis** [comment]
Might be fixed upstream by https://github.com/systemd/systemd/pull/5558

**Comment by dongsupark** [comment]
Yeah, maybe it's fixed in systemd v233 or newer. I'm not sure what we could do about this issue. We already recommend users to upgrading systemd to v233, though Ubuntu 17.04 is still based on v232 by default.

**Comment by iaguis** [comment]
It seems it's in v234

```
$ git tag --contains 7357272ed1c2c7a139c9ecbc8f3b8f63f71dd0b0
v234
```

**Comment by alban** [comment]
I have this issue on Ubuntu 17.10 too, despite having systemd >= v234.

On the host (Ubuntu 17.10):
```
$ systemctl --version
systemd 234
```
On the node (CoreOS 1590.0.0):
```
kubespawndefault0 ~ # systemctl --version
systemd 235
```

Is there any workaround? I tried using `nameserver 8.8.8.8` in resolv.conf (and after that I can start docker containers) but still `kube-spawn start` freezes, maybe because I am not fast enough in editing resolv.conf before it is used.

**Comment by dongsupark** [comment]
@alban I already added a workaround in a pending PR https://github.com/kinvolk/kube-spawn/pull/177. What I did was to disable the stub listener of systemd-resolved, which is the least invasive approach.

Or you can just stop and disable systemd-resolved. We cannot recommend it to every user, but it might be ok temporarily for a testing purpose.

AFAIK it's not enough to simply change `/etc/resolv.conf`.

**Comment by alban** [comment]
Setting milestone 0.2.0 to get at least the documentation explaining how to disable systemd-resolved on Ubuntu 17.10.

**Comment by alban** [comment]
Fixed by #177:
- documented in [distro.md](https://github.com/kinvolk/kube-spawn/blob/master/doc/distro.md#systemd-resolved)
- vagrant does the correct thing in [vagrant-mod-env.sh#L48-L59](https://github.com/kinvolk/kube-spawn/blob/master/scripts/vagrant-mod-env.sh#L48-L59)
